libxml2 2.9.14

owine opened the pull request:
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

chenrui333 commented:
looks like only a bunch of failures on linux build. how should we proceed on this?

chenrui333 commented:
@danielnachun can you weigh in about the linux build failures?

danielnachun commented:
Here's what I can see so far:
`ansible@2.9` - It looks it can't find the Python interpreter in `python@3.9` - it probably used to work up to `python@3.7`, but it doesn't look for `python3.8` and above.  
`gst-libav` - although this is already supposed to be built with GCC 11, it seems that `libgstlibav.so` is incorrectly using the host `libstdc++` instead.
`hbase` - not sure what's going on here, I just fixed this a few weeks ago and it worked fine.
`jetty` - this is known issue with our Docker setup, not the formula.
`samba` - needs `uses_from_macos "libxcrypt"` to be added
`solr@7.7` - no idea what's going on there
`streamlink` - the test URL now returns a 403 error, so a different one needs to be used
`stubby` - not sure here either
`zero-install` - similar to `ansible@3.9` it seems like it can't find the path to Python 3.

The linkage failures referencing `libcrypt` need the same fix of adding `uses_from_macos "libxcrypt"`.  The linkage failures about `postgresql`, `mysql` and `poppler` are spurious - they are happening because those formulae depend on `libpq`, `mysql-client` and `poppler-qt5`, respectively and linkage checker does not seem to recognize that the dependencies are already covered.

I think we should try to fix the `libcrypt` failures because the fix is trivial.  `gst-libav` may just be an RPATH issue, and  `ansible@2.9`, `zero-install`, and `streamlink` may not be too hard either.  The rest I think are trickier and not worth holding up the PR unless someone can figure out them quickly.

owine commented:
> Here's what I can see so far: `ansible@2.9` - It looks it can't find the Python interpreter in `python@3.9` - it probably used to work up to `python@3.7`, but it doesn't look for `python3.8` and above. `gst-libav` - although this is already supposed to be built with GCC 11, it seems that `libgstlibav.so` is incorrectly using the host `libstdc++` instead. `hbase` - not sure what's going on here, I just fixed this a few weeks ago and it worked fine. `jetty` - this is known issue with our Docker setup, not the formula. `samba` - needs `uses_from_macos "libxcrypt"` to be added `solr@7.7` - no idea what's going on there `streamlink` - the test URL now returns a 403 error, so a different one needs to be used `stubby` - not sure here either `zero-install` - similar to `ansible@3.9` it seems like it can't find the path to Python 3.
> 
> The linkage failures referencing `libcrypt` need the same fix of adding `uses_from_macos "libxcrypt"`. The linkage failures about `postgresql`, `mysql` and `poppler` are spurious - they are happening because those formulae depend on `libpq`, `mysql-client` and `poppler-qt5`, respectively and linkage checker does not seem to recognize that the dependencies are already covered.
> 
> I think we should try to fix the `libcrypt` failures because the fix is trivial. `gst-libav` may just be an RPATH issue, and `ansible@2.9`, `zero-install`, and `streamlink` may not be too hard either. The rest I think are trickier and not worth holding up the PR unless someone can figure out them quickly.

I've addressed the `libcrypt` failures here.

branchvincent commented:
looks like tests didn't run here?

owine commented:
Hoping tests can be run on this given it has several linkage changes

bayandin commented on Formula/postgresql@10.rb:
```diff
@@ -4,6 +4,7 @@ class PostgresqlAT10 < Formula
   url "https://ftp.postgresql.org/pub/source/v10.21/postgresql-10.21.tar.bz2"
   sha256 "d32198856d52a9a6f5d50642ef86687ac058bd6efca5c9ed57be7808496f45d1"
   license "PostgreSQL"
+  revision 2
```
```suggestion
  revision 1
```

owine replied:
Thanks - will commit this after CI finishes

alebcay commented:
I think `samba` may fail - there's an issue with the test block (see #101042). I suspect both the linkage fix and test block fix need to be picked from that PR into here.

chenrui333 commented:
reorg the commits and dropped the change for samba as it got fixed already. 🤞 for this run

chenrui333 commented:
on 12, 12-arm

```
brew install --verbose --build-bottle mailutils
```

SMillerDev commented:
I don't think this is supposed to happen and I'm curious why it's even working pre-12
```
  libtool: install: /usr/bin/install -c .libs/dotlock /usr/local/Cellar/mailutils/3.15_1/bin/dotlock
  /Applications/Xcode.app/Contents/Developer/usr/bin/make  install-exec-hook
  for i in dotlock; do\
  		chown root:mail /usr/local/Cellar/mailutils/3.15_1/bin/$i;\
  		chmod 2755 /usr/local/Cellar/mailutils/3.15_1/bin/$i;\
  	done
  chown: /usr/local/Cellar/mailutils/3.15_1/bin/dotlock: Operation not permitted
  chmod: Unable to change file mode on /usr/local/Cellar/mailutils/3.15_1/bin/dotlock: Operation not permitted
```

Bo98 commented:
Honestly, you really don't need to worry about `libxcrypt` in this PR - it's making it quite large. The linkage wasn't failing - it's a deprecation warning, not an error. It is absolutely not necessary to fix before merging this.

If you're nearly there: great, but just drop anything else for a separate PR.

> I don't think this is supposed to happen and I'm curious why it's even working pre-12
> 
> ```
>   libtool: install: /usr/bin/install -c .libs/dotlock /usr/local/Cellar/mailutils/3.15_1/bin/dotlock
>   /Applications/Xcode.app/Contents/Developer/usr/bin/make  install-exec-hook
>   for i in dotlock; do\
>   		chown root:mail /usr/local/Cellar/mailutils/3.15_1/bin/$i;\
>   		chmod 2755 /usr/local/Cellar/mailutils/3.15_1/bin/$i;\
>   	done
>   chown: /usr/local/Cellar/mailutils/3.15_1/bin/dotlock: Operation not permitted
>   chmod: Unable to change file mode on /usr/local/Cellar/mailutils/3.15_1/bin/dotlock: Operation not permitted
> ```

I've seen a couple of these on 12.3, but we never figured it out before we updated non-ephemeral. Happens with httpd too I think.

owine commented:
> 

Thanks for the feedback - `root` is being addressed elsewhere so it will need a drop here, but my thought is to leave the rest as I believe all else are ready to merge.

owine commented:
Sorry for a second comment. I’ll await direction before dropping `root` to be mindful of CI.

Bo98 commented:
> I believe all else are ready to merge.

Well, there's a few which fail to build on Linux + one on Monterey.

iMichka commented:
@owine you can remove `root` from this PR as #102135 has been merged. That should speed up the build slightly.

chenrui333 commented:
for `mailutils`, we need to patch the build step or we can drop `dotlock` utility build with `disable-build-dotlock`

chenrui333 commented:
The linux failure as below, we can followup on these cc @Homebrew/linux 
```
brew install antlr4-cpp-runtime
brew test --retry --verbose gst-libav
brew test --retry --verbose hbase
brew test --retry --verbose pdfsandwich
brew test --retry --verbose root
brew test --retry --verbose streamlink
brew test --retry --verbose stubby
```

For `antlr4-cpp-runtime`, install stats

```
==> Analytics
install: 71 (30 days), 344 (90 days), 1,742 (365 days)
```
